File: DataBaseRoomApp/jroomapp/src/main/java/com/woowrale/jroomapp/data/repository/LocalRepository.java
package com.woowrale.jroomapp.data.repository;

import com.woowrale.jroomapp.data.datasource.LocalDataSource;
import com.woowrale.jroomapp.domain.model.Contact;

import java.util.List;

public class LocalRepository {

    private LocalDataSource localDataSource;

    public LocalRepository(LocalDataSource localDataSource){
        this.localDataSource = localDataSource;
    }

    public Contact findById(Integer id){
        return localDataSource.findById(id);
    }

    public List<Contact> getAll(){
        return localDataSource.getPopularContacts();
    }

    public void saveAll(List<Contact> contacts){
        localDataSource.saveContacts(contacts);
    }

    public void updateContact(Contact contact){
        localDataSource.update(contact);
    }
}

File: DataBaseRoomApp/kroomapp/src/main/java/com/woowrale/kroomapp/data/repository/LocalRepository.kt
package com.woowrale.kroomapp.data.repository

import com.woowrale.kroomapp.data.datasource.LocalDataSource
import com.woowrale.kroomapp.domain.model.Contact

class LocalRepository (private val localDataSource: LocalDataSource) {

    fun findById(id: Int): Contact {
        return localDataSource.findById(id)
    }

    fun getAll(): List<Contact> {
        return localDataSource.getPopularContacts()
    }

    fun saveAll(contacts: List<Contact>) {
        localDataSource.saveContacts(contacts)
    }

    fun updateContact(contact: Contact) {
        localDataSource.update(contact)
    }
}